plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -510,6 +515,49 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
             // This is just a catch-all solution so that we have any actionable information when eg. SELECT * FROM information_schema.columns fails.
             throw new RuntimeException("Failed to construct table metadata for table " + tableName, e);
         }
+        return Optional.ofNullable(getAvroSchemaUrl(tableMetadata.getProperties()))
+                .map(a -> rewriteMetadataFromAvroSchema(session, tableMetadata, a))
+                .orElse(tableMetadata);
+    }
+
+    private ConnectorTableMetadata rewriteMetadataFromAvroSchema(ConnectorSession session, ConnectorTableMetadata tableMetadata, String avroSchemaUrl)
```

Comment:
If we are going to specify the new column details, shouldn't it be reflected in `HiveMetadata#getColumnHandles` too ?

Reply:
As discussed possible moving to BridgingHiveMetastore#getTable is best approach.